jax/_src/numpy/lax_numpy.py

```diff
@@ -6397,8 +6400,27 @@ def trim_zeros(filt, trim='fb'):
     >>> jnp.trim_zeros(x)
     Array([2, 0, 1, 4, 3], dtype=int32)
   """
-  filt = core.concrete_or_error(asarray, filt,
+  if isinstance(filt, list):
+    # Added on Sep 11, 2024
+    if deprecations.is_accelerated("jax-numpy-trimzeros-lists"):
+      util.check_arraylike("trim_zeros", filt)
+    deprecations.warn(
+      "jax-numpy-trimzeros-lists",
+      "Passing python lists to jnp.trim_zeros() is deprecated, since JAX APIs doesn't"
+      "work on Python Lists. Currently, it works with Python Lists. In the future this"
+      "will result in an error.",
+      stacklevel=2)
+  filt = core.concrete_or_error(jax.numpy.asarray, filt,
```

Comment:
This construct won't work well with type checking. I'd suggest doing this:
```python
core.concrete_or_error(None, filt, "Error arose...")
filt_arr = jax.numpy.asarray(filt)
del filt  # prevents accidental use of the old name
```
and then work with `filt_arr` below so that mypy won't complain below. The problem with the current approach is that `filt` is annotated as `ArrayLike`, and `ArrayLike` doesn't have `ndim` properties, etc.

Reply:
Thanks! Updated with suggested changes